### ISSUE TITLE
[Explorer] Remove H5AD and conversion code

### ISSUE THREAD

**Issue by MDunitz** [chanzuckerberg/single-cell-data-portal#1277]
## Need
- The explorer will only support the CXG file format

## Approach
- remove conversion code
- refactor data_common and and data_cxg to be one class, remove the factory pattern
- Remove `AnndataAdaptor` class and its usages.
- Remove peripheral code as well, such as `diffexp_generic.py` that is related to processing anndata files.
- Delete tests accordingly.
- Fix the unit tests and smoke tests to use CXG test files instead of H5AD test files.
- Delete H5AD fixtures from the test directory.
- update documentation to remove references non cxg file formats

## DOD
- The explorer repo only supports/references the cxg file format

**Why**: the hosted server does not need to support reading from H5AD anymore since the local server has been split out to be able to read from H5AD. We only store CXGs for the hosted server to read on the backend and that is the plan moving forward.

**Comment by MDunitz** [comment]
Look out for hard coding in unexpected places
It may be necessary to regenerate a dataset in the correct format (specifically the pbmc3k dataset)

**Comment by atolopko-czi** [comment]
TLDR: If anyone is taking this on, talk with me first. I may have done most of the work. 

I ended diving into this rabbit hole of a story, as I continued to learn the code base more in depth while debugging Mac M1 package installation issues, starting from "Do we actually need this package that won't install on M1?". Have since removed H5AD/AnnData and user annotations (including substantial amount of config-related code for user annotations). The changes go deep and wide, but I think my latest code is valid. Local Explorer runs.  Tests still need work. Need to double check many things with team to ensure I haven't ripped out too much.

**Comment by atolopko-czi** [comment]
Addressed by https://github.com/chanzuckerberg/single-cell-explorer/pull/180
